django/db/models/sql/compiler.py

```diff
@@ -146,9 +143,21 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr, 'alias', None) == self.query.tables[0]):
                     pk = expr
                     break
+            # The logic here is: if the main model's primary key is in the
```

Comment:
It's hard to say, it was a big refactoring so maybe just a typo, or in some preliminary version we had `new_expressions` instead of `expressions`.